src/components/common/edge/rectangularEdge.tsx

```diff
@@ -1,119 +1,271 @@
         // y will remain same and x will vary
-        let offset = isToRight ? 10 : -10;
-        let midPoint = { x: offset + (start.x + end.x) / 2, y: start.y };
-        let pointA = { x: isToRight ? midPoint.x - 10 : midPoint.x + 10, y: start.y };
-        let pointB = { x: isToRight ? pointA.x - 10 : pointA.x + 10, y: start.y };
-        let pointC = { x: pointB.x, y: start.y - 10 };
-        let pointD = { x: pointB.x, y: start.y + 10 };
-
-        return `M${midPoint.x} ${midPoint.y} L ${pointC.x} ${pointC.y} L ${pointA.x} ${pointA.y} L ${pointD.x} ${pointD.y} Z`;
+        let offset = isToRight ? 10 : -10
+        let midPoint = { x: offset + (start.x + end.x) / 2, y: start.y }
+        let pointA = { x: isToRight ? midPoint.x - 10 : midPoint.x + 10, y: start.y }
+        let pointB = { x: isToRight ? pointA.x - 10 : pointA.x + 10, y: start.y }
+        let pointC = { x: pointB.x, y: start.y - 10 }
+        let pointD = { x: pointB.x, y: start.y + 10 }
+
+        return `M${midPoint.x} ${midPoint.y} L ${pointC.x} ${pointC.y} L ${pointA.x} ${pointA.y} L ${pointD.x} ${pointD.y} Z`
     }
 
     getArrowEquation(): string {
-        let { lineStartX, lineStartY, lineEndX, lineEndY, midPointX, midPointY } = this.getLineDots();
+        let { lineStartX, lineStartY, lineEndX, lineEndY, midPointX, midPointY } = this.getLineDots()
         if (lineStartX > lineEndX) {
             // this.props.deleteEdge();
-            return this.getSegmentArrowEquationHorizontal({
-                x: midPointX,
-                y: midPointY
-            }, {
+            return this.getSegmentArrowEquationHorizontal(
+                {
+                    x: midPointX,
+                    y: midPointY,
+                },
+                {
                     x: lineStartX,
-                    y: lineEndY
-                }, lineStartX < lineEndX);
-        }
-        else {
-            return this.getSegmentArrowEquationHorizontal({
-                x: midPointX,
-                y: lineEndY
-            }, {
+                    y: lineEndY,
+                },
+                lineStartX < lineEndX,
+            )
+        } else {
+            return this.getSegmentArrowEquationHorizontal(
+                {
+                    x: midPointX,
+                    y: lineEndY,
+                },
+                {
                     x: lineEndX,
-                    y: lineStartY
-                }, lineStartX < lineEndX);
+                    y: lineStartY,
+                },
+                lineStartX < lineEndX,
+            )
+        }
+    }
+
+    getPipelineType = () => {
+        if (this.props.isWebhookCD) {
+            return PipelineType.WEBHOOK
+        }
+
+        if (this.props.startNode.type === WorkflowNodeType.CI) {
+            return PipelineType.CI_PIPELINE
+        }
+
+        return PipelineType.CD_PIPELINE
+    }
+
+    handleAddCD = (position: AddCDPositions) => {
```

Comment:
extract this common code so that it can be reuse

Reply:
Done